### PR TITLE
employee service completed

### PR DESCRIPTION
**Pull request by amankagarwal**

### CODE REVIEW

File: src/main/java/com/careerfair/q/model/db/Employee.java [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
-@Getter
-@Setter
+@AllArgsConstructor
+@NoArgsConstructor
 public class Employee {
 
     @PropertyName("employee_id")
-    @JsonProperty("employee_id")
+    @Exclude
     public String employeeId;
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/model/db/Employee.java:17]: private fields

File: src/main/java/com/careerfair/q/service/employee/response/EmployeeResponse.java [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
 
+import com.careerfair.q.model.db.Employee;
+import com.careerfair.q.util.enums.Role;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
 public abstract class EmployeeResponse {
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/service/employee/response/EmployeeResponse.java:11]: Why don't you use @Data here

File: src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java [hunk 3]
```diff
@@ -85,14 +86,50 @@ public void registerStudentToEmployee(String employeeId, String studentId)
...
+        }
+    }
+
+    @Override
+    public Employee addEmployee(Employee newEmployee) throws FirebaseException {
+        Firestore firestore = FirestoreClient.getFirestore();
+        DocumentReference documentReference = firestore.collection(EMPLOYEE_COLLECTION).document();
+        String employeeId = documentReference.getId();
+
+        // TODO: this currently will also add a field students and set that to null
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java:119]: There is  a TODO here

File: src/test/java/com/careerfair/q/service/employee/implementation/EmployeeServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        assertEquals(newStudent.bio, addEmployeeRequest.getBio());
+        assertEquals(newStudent.email, addEmployeeRequest.getEmail());
+        assertEquals(newStudent.role, addEmployeeRequest.getRole());
+
+        doReturn(employee).when(firebaseService).addEmployee(any());
+
+        AddEmployeeResponse addEmployeeResponse = employeeService.addEmployee(addEmployeeRequest);
+
+        // students is not an attribute in response or request
+        assertNotNull(addEmployeeResponse);
```
Comment (SarvagyaGupta) [inline src/test/java/com/careerfair/q/service/employee/implementation/EmployeeServiceTest.java:190]: Make these checks into one private method

File: src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java (hunk 3 above)
Comment (amankagarwal) [reply src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java:119]: yes leaving this for now

File: src/test/java/com/careerfair/q/workflow/database/EmployeeFirebaseWorkflowTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    private DocumentSnapshot documentSnapshot;
+
+    @InjectMocks
+    private final EmployeeFirebaseWorkflowImpl employeeFirebaseWorkflow =
+            new EmployeeFirebaseWorkflowImpl();
+
+    @BeforeEach
+    public void setupMock() {
+        MockitoAnnotations.initMocks(this); // enable mockito annotations
+    }
```
Comment (SarvagyaGupta) [inline src/test/java/com/careerfair/q/workflow/database/EmployeeFirebaseWorkflowTest.java:50]: Ummm... This test suite is not complete

Comment (amankagarwal) [reply src/test/java/com/careerfair/q/workflow/database/EmployeeFirebaseWorkflowTest.java:50]: will remove from this PR

File: src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java (hunk 3 above)
Comment (amankagarwal) [reply src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java:119]: for reference later if issues come while testing end to end

Comment (SarvagyaGupta) [reply src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java:119]: It doesn't seem like a big task. Is it possible to do it in this PR?

Comment (amankagarwal) [reply src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java:119]: yes- do you want to not include students as a null field when adding a new employee? Or is it fine if students is set to null (both cases would work technically)

**Pull request merged**